app/controllers/admin/api_controller.rb

```diff
@@ -5,18 +5,29 @@ class Admin::ApiController < Admin::AdminController
   # If we used "api_key", then our user provider would try to use the value for authentication
 
   def index
-    keys = ApiKey
-      .where(hidden: false)
-      .includes(:user, :api_key_scopes)
+    offset = (params[:offset] || 0).to_i
+    limit = (params[:limit] || 50).to_i.clamp(1, 50)
```

Comment:
Minor but I'm not sure if the limit needs to be customizable by the client.

Reply:
I think we use this pattern in more places. I wanted to be consistent.